nrf-hal-common/src/pwm.rs

```diff
@@ -748,8 +748,46 @@ where
     }
 
     /// Consumes `self` and returns back the raw peripheral.
-    pub fn free(self) -> T {
-        self.pwm
+    pub fn free(
+        self,
+    ) -> (
+        T,
+        Option<Pin<Output<PushPull>>>,
+        Option<Pin<Output<PushPull>>>,
+        Option<Pin<Output<PushPull>>>,
+        Option<Pin<Output<PushPull>>>,
```

Comment:
It probably makes sense to create a `Pin` struct in every module and return that, unless the module only deals with a single pin. Otherwise these signatures get a bit unwieldy.

Reply:
Makes sense, I'll go through and add `Pins` structs where none existed yet.